ipfs_api_mount/__init__.py

```diff
@@ -155,13 +152,15 @@ def _is_raw_block(self, cid):
 
         return cid_bytes.startswith(bytes([0x01, 0x55]))
 
+    @lru_cache(maxsize=1024 * 1024)
```

Comment:
`path_is_dir`, `path_is_file` and `path_size` look fast - they are simple and without any IO operations (excluding subfunctions). I wouldn't do caching for them.

I guess problem with slow listing is solved by increasing cache on `resolve_path` and `block_size` solely.

Reply:
Halt, thats not true what i said. They must be cached, because of not sufficient caching on object_data.

I feel that solution is not good because, like you said, it's just masking the problem. I have to think and try some changes. Recently I've found `kernel_cache` and `auto_cache` fuse options. There is also low-level fuse api which (AFAIK) would allow kernel to manage the cache on the ipfs-api-mount side (by creating/forgetting inodes). It may be a way to go.

Reply:
As I wrote in #12 we should add kernel caching additionally. There's zero issue with doing both. But we shouldn't do any 'data' caching, since IPFS stores everything in it's cache anyway, which is also cached by the kernel.

I would reduce the cache to metadata caching. Adding a useful raw-data caching would require a request counter, memory pressure analyzation and more. I think that's out of the scope of this python program - especially since we already got good implementations in the kernel for this.